File: submissions/nightmare-builders/8.java
	static void f(String s, int n, int width) {
		int rows = (int) java.lang.Math.ceil(((double)s.length()) / (n * width));
		
		for (int row = 0; row < rows; row++) {
			for (int i = 0; i < n; i++) {
				for (int j = 0; j < width; j++) {				
					int index = rows * i * width + row * width + j;					
					System.out.print(index < s.length() ? s.charAt(index) : " ");					
				}
				if (i < n-1) {
					System.out.print("|");
				}
			}
			System.out.println();
		}
	}

File: submissions/nightmare-builders/8.rb
def f(s, n, width)
		rows = (s.length.to_f / (n * width)).ceil
		for row in 0..rows-1
		
    		for i in 0..n-1
    		    for j in 0..width-1
					index = rows * i * width + row * width + j					
					p(index < s.length() ? s[index] : " ")
                    
				end
				if (i < n-1) 
					p( "|")
                end
		    end
			puts
		end
		
end

def p(str)
  print str
  $stdout.flush
end